### PR TITLE
Remove intermediate sync for test build

### PR DESCRIPTION
**Pull request by karajas**
This should fix the build break in official test builds.
@jashook @wtgodbe @weshaggard

### CODE REVIEW

**Comment by karajas** [comment]
@dotnet-bot test Ubuntu x64 Innerloop Formatting please

**Comment by jashook** [comment]
Looks like an infra error on the format job feel free to ignore. You have not touched any of the jit sources anyways.

File: buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json [hunk 1]
```diff
@@ -41,24 +41,6 @@
       "enabled": true,
       "continueOnError": false,
       "alwaysRun": false,
-      "displayName": "Sync test native binaries",
```
Comment (weshaggard) [inline buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json]: How does this relate to the intermedateAzureFeed issue?

File: tests/dir.props [hunk 2]
```diff
@@ -89,8 +89,8 @@
       https://dotnet.myget.org/F/dotnet-corefxlab/api/v3/index.json;
       $(RestoreSources)
     </RestoreSources>
-    <RestoreSources Condition="'$(OverwriteCoreClrPackageVersion)' == 'true'">
-      $(PackagesDir)AzureTransfer\;
+    <RestoreSources Condition="'$(IntermediateAzureFeed)' != ''">
```
Comment (weshaggard) [inline tests/dir.props]: Thanks for fixing this merge issue caused by my change into master.

File: buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json (hunk 1 above)
Comment (karajas) [reply buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json]: Reverting this change, thought this was syncing packages.

**Pull request merged**